### DISCUSSION TITLE
BFF Works in Dev but Not Production

### DISCUSSION THREAD

**Discussion by bmoellers** [web-infra-dev/modern.js#6428] [Q&A]
So sorry, I will see if I can write up a simple example to reproduce it when I can but I am seeing a weird problem.

I enabled BFF using the code here: https://modernjs.dev/en/guides/advanced-features/bff/function.html#bff-functions. Basically I added the same hello route:

```
export const get = async () => 'Hello Modern.js';
```

I ran the new to add the BFF things.

```
    moduleFederationPlugin(),
    expressPlugin(),
    bffPlugin(),
```

I did also add an `api/app.ts` just to see any access.

```
import express from 'express';

const app = express();

app.use(async (req, res, next) => {
  console.info(`access url: ${req.url}`);
  next();
});


export default app;
```

This all works great running `modern dev`. Can access the API directly in the browser. Can access the pages in the browser.

However when I do `modern build` then `modern serve` which are the production builds mentioned at https://modernjs.dev/en/apis/app/commands.html#modern-build and https://modernjs.dev/en/apis/app/commands.html#modern-serve. However I just get a `cannot get` error for the hello api route. When typing the wrong route I get a 404 so even though the api returns a 404 in the network tab from a page it seems to be different than a normal 404.

I then just tried `modern deploy` and then `PORT=3200 node .output/index.js` and same thing. Gets the error `Cannot GET /api/hello`. Pages still load.

When I look at the `.output` from the deploy I see an `api/lambda` directory but no files in it. Same for the `dist` folder from the build.

I swear this was working before but I don't see any changes that should be breaking this. The app.ts never fires either. Any ideas why this might be?

**Comment by julia-otran** [comment]
I had this problem today.

To me, removing 'noEmit' and adding ```"include": "api/**/*.ts"' ``` to tsconfig.json fixed the problem.
